pkg/skaffold/build/prebuilt.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2018 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package build
+
+import (
+	"context"
+	"io"
+
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/build/tag"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/constants"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/docker"
+	"github.com/GoogleContainerTools/skaffold/pkg/skaffold/schema/latest"
+	"github.com/pkg/errors"
+)
+
+// NewPreBuiltImagesBuilder returns an new instance a Builder that assumes images are
+// already built with given fully qualified names.
+func NewPreBuiltImagesBuilder(images []string) Builder {
+	return &prebuiltImagesBuilder{
+		images: images,
+	}
+}
+
+type prebuiltImagesBuilder struct {
+	images []string
+}
+
+// Labels are labels applied to deployed resources.
+func (b *prebuiltImagesBuilder) Labels() map[string]string {
+	return map[string]string{
+		constants.Labels.Builder: "pre-built",
+	}
+}
+
+func (b *prebuiltImagesBuilder) Build(ctx context.Context, out io.Writer, tagger tag.Tagger, artifacts []*latest.Artifact) ([]Artifact, error) {
+	tags := make(map[string]string)
```

Comment:
Since we’re already here, how about we generate the git tags if no images were supplied?

Reply:
We can also resolve the taggers from the yams if you want to be really correct (to make sure we can generate reproducibly)

Reply:
Sorry @r2d4, I don't understand

Reply:
If images isn't provided, try to seed it by using the passed in tagger to generate a tag for each artifact
```golang
if images == nil {
  for _, a := range artifacts { 
    tag, err := tag.GenerateFullyQualifiedTag(a.Image, nil)
    if err != nil // this is the case where we ended up with a tagger that needs some options passed in, fail
    images == append(images, tag)
}

Reply:
which would fix https://github.com/GoogleContainerTools/skaffold/issues/653